### PR TITLE
Mixer: fix channel ordering issues and fix display of channels from instrument changes

### PR DESCRIPTION
**Pull request by cbjeukendrup**
Resolves: #11199
Resolves: #10815

To do:
- a mysterious crash occurs when adding an instrument change

### CODE REVIEW

**Comment by Eism** [comment]
@bkunda 
Please check this PR

**Comment by bkunda** [comment]
Tested on MacOS 🟢
(Was unable to install on Windows...kept getting a message "The application was unable to start correctly (0xc0000106)")

On MacOS, everything seems to be working as expected. All issues in #10815 were resolved, and I had success inserting instrument changes too, so that resolves #11199 too.

**Comment by Eism** [comment]
@cbjeukendrup, can you help Bradley with Windows?

**Comment by cbjeukendrup** [comment]
I don't experience any problems if I try this build on Windows... maybe re-downloading helps?

**Comment by DmitryArefiev** [comment]
@cbjeukendrup I tested this on **Windows** and **Linux** - #10815 looks fine, but **with #11199 I had a crashes** in some cases:



https://user-images.githubusercontent.com/90187801/182902889-20ecc3ad-6dd0-4463-abd4-17aff609596a.mp4

Can you try this on your side?

1. Open [audio_Basic.zip](https://github.com/musescore/MuseScore/files/9261559/audio_Basic.zip)
2. Select the 2nd measure in Violin I
3. Change instrument to Clarinet

**Comment by cbjeukendrup** [comment]
@DmitryArefiev Thanks for testing! Yes, I can now also reproduce the crash again. I'm trying to find out what's going on exactly, but it's not obvious so far!

**Comment by cbjeukendrup** [comment]
@DmitryArefiev I changed something that fixes the crash for me and seems to have no side effects (all parameters in the Mixer still update properly if you change them). 

@vpereverzev That change is in the last commit. I'm not very confident that it is correct, but it seems to work. Also, Asan reported:
```
heap-use-after-free channel.h:126 in deto::async::Channel<QAccessibleEvent*>::ChannelInvoker::doInvoke(int, void*, deto::async::NotifyData const&)
```
and the source of this use-after-free seemed to be in the async library when `AsyncMode::AsyncSetRepeat` is used.

File: src/playback/qml/MuseScore/Playback/internal/MixerSoundSection.qml [hunk 1]
```diff
@@ -47,7 +47,7 @@ MixerPanelSection {
             height: 26
 
             supportsByPassing: false
-            resourceItemModel: channelItem.inputResourceItem
+            resourceItemModel: channelItem.inputResourceItem ?? null
```
Comment (Eism) [inline src/playback/qml/MuseScore/Playback/internal/MixerSoundSection.qml:50]: ohh

**Review by Eism — APPROVED** [review]

File: src/playback/qml/MuseScore/Playback/internal/MixerSoundSection.qml (hunk 1 above)
Comment (cbjeukendrup) [reply src/playback/qml/MuseScore/Playback/internal/MixerSoundSection.qml:50]: @Eism The bug is fixed in Qt Creator 8 :)

**Pull request merged**